### PR TITLE
Clowder integration final

### PR DESCRIPTION
**Pull request by bennyturns**
Signed-off-by: Ben Turner <bturner@redhat.com>

## Link(s) to Jira
- https://issues.redhat.com/browse/RHCLOUD-10213

## Description of Intent of Change(s)
This change is to enable clowder support for RBAC.

## Local Testing
How can the feature be exercised?   Normal deploy / smoke test / qe testing in ci, qa, stage, and prod as well as the new ephemeral envs.

How can the bug be exploited and fix confirmed?  See above.

Is any special local setup required?  Yes, this should be tested as an ephemeral env as well as stage / prode / ci / qa.

## Checklist
- [ * ] if API spec changes are required, is the spec updated?

- [ * ] are there any pre/post merge actions required? if so, document here.
- We will need to deploy to all ENVs and test.

- [ * ] are theses changes covered by unit tests?

- [ * ] if warranted, are documentation changes accounted for?

- [ ] does this require migration changes?
  - [ ] if yes, are they backwards compatible?

- [ ] is there known, direct impact to dependent teams/components?
  -  if yes, how will this be handled?

## Secure Coding Practices Checklist Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Practices Checklist
- [ ] Input Validation

- [ ] Output Encoding

- [ ] Authentication and Password Management

- [ ] Session Management

- [ ] Access Control

- [ ] Cryptographic Practices

- [ ] Error Handling and Logging

- [ ] Data Protection

- [ ] Communication Security

- [ ] System Configuration

- [ ] Database Security

- [ ] File Management

- [ ] Memory Management

- [ ] General Coding Practices

### CODE REVIEW

File: openshift/s2i/bin/run [hunk 1]
```diff
@@ -80,6 +80,7 @@ if should_migrate; then
     echo "WARNING: seems that you're using Django, but we could not find a 'manage.py' file."
     echo "Skipped 'python manage.py migrate'."
   fi
+  exit 0
```
Comment (bennyturns) [inline openshift/s2i/bin/run:83]: We need this so that the init container finishes after the DB upgrade FYI

File: rbac/rbac/settings.py [hunk 2]
```diff
@@ -215,10 +216,24 @@
     "ORDERING_PARAM": "order_by",
 }
 
-CW_AWS_ACCESS_KEY_ID = ENVIRONMENT.get_value("CW_AWS_ACCESS_KEY_ID", default=None)
-CW_AWS_SECRET_ACCESS_KEY = ENVIRONMENT.get_value("CW_AWS_SECRET_ACCESS_KEY", default=None)
-CW_AWS_REGION = ENVIRONMENT.get_value("CW_AWS_REGION", default="us-east-1")
-CW_LOG_GROUP = ENVIRONMENT.get_value("CW_LOG_GROUP", default="platform-dev")
+# CW settings
+if ENVIRONMENT.bool("CLOWDER_ENABLED", default=False):
+    if ENVIRONMENT.bool("CW_NULL_WORKAROUND", default=True):
```
Comment (bennyturns) [inline rbac/rbac/settings.py:221]: This is for the null CW workaround

File: deploy/clowd-environment.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
+---
```
Comment (psav) [inline deploy/clowd-environment.yml:1]: I'm not sure you need the clowd-environment here at all

File: deploy/rbac-clowdapp.yml [hunk 4]
```diff
@@ -0,0 +1,555 @@
...
+metadata:
+  name: rbac
+objects:
+- apiVersion: cloud.redhat.com/v1alpha1
+  kind: ClowdApp
+  metadata:
+    name: rbac
+  spec:
+    envName: ${ENV_NAME}
+    pods:
```
Comment (psav) [inline deploy/rbac-clowdapp.yml:13]: We should move this to newer deployments style

**Comment by psav** [comment]
The only other piece I would say on this one is inside the s2i run script it hard codes the port to 8000, this matches what we want, but if we were ever to change that in another clowdenvironment, the service would fail

**Review by coderbydesign — APPROVED** [review]
Approving based on the RBAC-specific changes. 👍

File: deploy/clowd-environment.yml (hunk 3 above)
Comment (bennyturns) [reply deploy/clowd-environment.yml:1]: OK I will remove it

**Comment by kylape** [comment]
```
rbac/rbac/database.py:19:1: I100 Import statements are in the wrong order. 'from app_common_python import LoadedConfig' should be before 'from .env import ENVIRONMENT' and in a different group.
rbac/rbac/database.py:19:1: I201 Missing newline between import groups. 'from app_common_python import LoadedConfig' is identified as Third Party and 'from .env import ENVIRONMENT' is identified as Application Relative.
rbac/rbac/database.py:21:1: E302 expected 2 blank lines, found 1
```

File: deploy/rbac-clowdapp.yml (hunk 4 above)
Comment (bennyturns) [reply deploy/rbac-clowdapp.yml]: done!

**Pull request merged**